### ISSUE TITLE
support select

### ISSUE THREAD

**Issue by roboojack** [rosmo/sails-arangodb#6]
Hi, would it be difficult to implement select functionality?

**Comment by jhawlwut** [comment]
+1

**Comment by rosmo** [comment]
You mean this: https://github.com/balderdashy/waterline/issues/73 ?

**Comment by jhawlwut** [comment]
Yes, that's it

I was able to build an options.select function;however, I'm having trouble
getting it to transform into the AQL structure for 'RETURN'

It builds into something like {value: d.name}

On Friday, May 8, 2015, Taneli Leppä notifications@github.com wrote:

> You mean this: balderdashy/waterline#73
> https://github.com/balderdashy/waterline/issues/73 ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/rosmo/sails-arangodb/issues/6#issuecomment-100139784.

## 

Sent from my iPhone

**Comment by roboojack** [comment]
Yes, but send the RETURN [x.attribute1, x.attribute2, x.attribute3] part of the AQL to ArangoDB so that we don't have to filter the JSON that's returned. It's a query optimization, but it would be a nice feature.

**Comment by rosmo** [comment]
Can you guys give me an example how you'd like this to work? I'm using Model.query mostly for my queries (since most of my queries are GRAPH_NEIGHBORS lookups) and those have to specified completely manually.

**Comment by rosmo** [comment]
Can you try this branch: https://github.com/rosmo/sails-arangodb/tree/select-support

It (at least tries to) implements Model.find({ where : { ... }, select: [ 'field1', field2', ...] })  support.

**Comment by gabriel-letarte** [comment]
I will work on getting this feature into master.

**Comment by gbevan** [comment]
Is it still planned for where and select support to be merged/supported?
I'm using waterline from feathersjs - would be very nice to be able to select fields etc

**Comment by gbevan** [comment]
This pull request (in progress) implements select support - https://github.com/gabriel-letarte/sails-arangodb/pull/22
